File: wc_tool.py
from pathlib import Path
from check_file_permission import check_file_permission

def wc_file(file_path, config):
    if not check_file_permission(file_path, "read", config):
        raise PermissionError(f"Read access denied for {file_path}")
    file_path = Path(file_path).resolve()
    if not file_path.is_file():
        raise ValueError(f"File not found: {file_path}")
    with open(file_path, "r", encoding="utf-8") as f:
        content = f.read()
    lines = len(content.splitlines())
    words = len(content.split())
    chars = len(content)
    return f"{lines} {words} {chars} {file_path}"

File: packages/core/src/tools/wc-tool.ts
import { promises as fs } from 'fs';
import { resolve } from 'path';
import { checkFilePermission } from '../services/filePermissionService';

export async function wcFile(
  filePath: string,
  config: unknown
): Promise<string> {
  if (!(await checkFilePermission(filePath, 'read', config))) {
    throw new Error(`Read access denied for ${filePath}`);
  }

  const resolvedPath = resolve(filePath);
  const stats = await fs.stat(resolvedPath);

  if (!stats.isFile()) {
    throw new Error(`File not found: ${resolvedPath}`);
  }

  const content = await fs.readFile(resolvedPath, 'utf-8');
  const lines = content.split('\n').length;
  const words = content.split(/\s+/).filter(Boolean).length;
  const chars = content.length;

  return `${lines} ${words} ${chars} ${resolvedPath}`;
}